README.md

```diff
@@ -1,36 +1,121 @@
+```
+npm install && npm run install-idl && npm run generate:idl
+```
+2. Build the project files
+```
+npm run build
+```
+3. After building the code, the server can be started by running this command from the same directory that includes the build
+```
+npm start
+```
+4. The webapp can be accessed now through `localhost:8088` (port can be changed using `CADENCE_WEB_PORT` environment variable)
+
+#### Running development environment
+
+To run development server, follow this steps:
+
+1. Install npm packages and download idls
+```
+npm install && npm run install-idl && npm run generate:idl
+```
+2. Run development server using
+```
 npm run dev
-# or
-yarn dev
-# or
-pnpm dev
-# or
-bun dev
+```
+3. The webapp can be accessed now through `localhost:8088` (port can be changed using `CADENCE_WEB_PORT` environment variable)
+
+Note: For contribution we recommend using dev containers, check [VSCode Dev Containers](#using-vscode-dev-containers) section for more information
+
+#### Using VSCode Dev Containers
+
+1. Set up the [Remote Containers plugin](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) in VSCode.
+2. Open the cadence-web directory in VSCode.
+3. Make sure to update `CADENCE_GRPC_PEERS` with the correct host. (If you are connecting to a server on a container host machine use `host.docker.interal:7833`, where `7833` is the gRPC port for a running [cadence-frontend](https://github.com/cadence-workflow/cadence/tree/master/service/frontend) service)
+4. Use the Command Palette to select the 'Reopen folder in Container' option
+5. Follow same commands listed in [Running development environment](#running-development-environment) section.
+
+
+#### Developing cadence-web against cadence composed docker
+
+To start development against a dockerized cadence services, run the following command from the root of the project
+```
+docker-compose -f docker-compose-backend-services.yml up
 ```
 
-Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
+You can customize the YAML file or reuse configurations from the [cadence repository](https://github.com/cadence-workflow/cadence/tree/master/docker). (In case of reusing exsisting files: ensure that cadence-web is not included in the composed container services, or just remove it)
 
-You can start editing the page by modifying `app/page.tsx`. The page auto-updates as you edit the file.
+After running `cadence`, start `cadence-web` for development using one of the previous methods ([Running development environment](#running-development-environment),[VSCode Dev Containers](#using-vscode-dev-containers) )
 
-This project uses [`next/font`](https://nextjs.org/docs/basic-features/font-optimization) to automatically optimize and load Inter, a custom Google Font.
 
-## Learn More
+#### NPM scripts
 
-To learn more about Next.js, take a look at the following resources:
+`build`: `Generates production builds`
```

Comment:
The line breaks are not being rendered correctly here. You could use a table if that is easier